### PR TITLE
Create Modal component

### LINKED ISSUE #43: Generic Modal + I don't know modal
## Description

Clicking the "I don't know" button under Solar panel orientation in the panel details form should popup a modal with some more details. We need a generic modal component to be reused and the content for this specific instance. Along with that, please create a reusable button component.

image

## Todos

These points are a rough guideline. **Please** feel free to discuss with others on the team about the best way to design this component!

- [x] Create generic modal component
- [x] Create generic button (from modal button styles)
- [x] Use in specific instance for i don't know button
- [x] Open modal on click

## Useful links

- https://www.figma.com/file/Y2IIxfo496bwxFsz8zo9hI/Wireframes?node-id=425%3A2092&t=BV2V0vpvaxOq9A8r-0
- https://github.com/KittyGiraudel/react-a11y-dialog

### PR DESCRIPTION
**Pull request by ashayp22**
## Description

Create a controlled Modal component for a generic Modal + I don't know modal.  Visit [`\form`](https://pv-sites-mobile-lvsel9gy0-hack4impact1.vercel.app/form) to see the modal.

Fixes #43 

## Todos

None

### CODE REVIEW

File: components/Modal.tsx
Comment (AndrewLester) [inline components/Modal.tsx]: Since it's no longer for testing.

```suggestion
  const [show, setShow] = useState<boolean>(false);
```

File: pages/form.tsx
Comment (AndrewLester) [inline pages/form.tsx]: *THIS COMMENT is unrelated to this line*

Can you have this modal open when the "I don't know" button is clicked on the page.

**Review by AndrewLester — CHANGES_REQUESTED** [review]

File: pages/form.tsx
Comment (AndrewLester) [inline pages/form.tsx]:
```suggestion
  const [show, setShow] = useState<boolean>(false);
```

**Review by AndrewLester — APPROVED** [review]
One last thing, approving so you can merge once done. Thanks for doing this quick

File: components/Modal.tsx
Comment (ericcccsliu) [inline components/Modal.tsx]: could we change the width of the modal to line up with the widths of the form input fields behind it?

**Review by ericcccsliu — APPROVED** [review]
great work! left one comment—after that you're all good to merge.

**Pull request merged**
